### ISSUE TITLE
Client lib: failure to create ASD connection is not handled

### ISSUE THREAD

**Issue by redlicha** [openvstorage/alba#580]
Observed during tests: volumedriver receives exceptions from the client lib:
```
Jan 20 09:53:51 ftcmp01 volumedriver_fs.sh[26324]: 2017-01-20 09:53:51 646596 +0100 - ftcmp01 - 26324/0x00007fc32b948700 - volumedriverfs/AlbaProxyClient - 000000000005502a - info - logMessage: TCP_transport(172.22.186.32, 8611)
[...]
Jan 20 09:53:56 ftcmp01 volumedriver_fs.sh[26324]: 2017-01-20 09:53:56 636157 +0100 - ftcmp01 - 26324/0x00007fc328141700 - volumedriverfs/AlbaConnection - 0000000000055032 - error - convert_exceptions_: partial read: caught exception: Operation canceled
```

Code inspection suggests that exceptions during ASD client instantiation escape without falling back to the slow path.

**Comment by redlicha** [comment]
```
int OsdAccess::_read_osd_slices_asd_direct_path(
    osd_t osd, std::vector<asd_slice> &slices) {
[...]
  auto connection = p->get_connection();  

  if (connection) {
    try {
     [...] 
     p->release_connection(std::move(connection));
      return 0;
    } catch (std::exception &e) {
      ALBA_LOG(INFO, "exception in _read_osd_slices_asd_direct_path for osd "
                         << osd << " " << e.what());
```
^^^ Please log at ERROR level.

**Comment by toolslive** [comment]
fixed by ( #583)
